austrakka/components/project/dataset/__init__.py

```diff
@@ -0,0 +1,57 @@
+import click
+
+from austrakka.utils.options import opt_analysis_label, opt_tracking_token, opt_abbrev, opt_detailed
+from austrakka.components.project.dataset.funcs import add_dataset_blocking
+from austrakka.components.project.dataset.funcs import add_dataset, track_dataset
+from austrakka.utils.output import table_format_option
+
+
+@click.group()
+@click.pass_context
+def dataset(ctx):
+    """Commands to upload project datasets"""
+    ctx.context = ctx.parent.context
+
+
+@dataset.command('async-add')
+@click.option('-fp',
+              '--file-path',
+              help='dataset csv file to upload to the project', )
+@opt_analysis_label()
+@opt_abbrev(help="Project Abbreviation")
+def dataset_add(
+        file_path: str,
+        analysis_label: str,
+        abbrev: str):
+    """Upload a dataset file to the given project in AusTrakka."""
+    add_dataset(file_path, analysis_label, abbrev)
+
+
+@dataset.command('track-job')
+@opt_abbrev(help="Project Abbreviation")
```

Comment:
I'm wary of calling these "jobs" in a user-facing way (i.e. the command name track-job). I think for binfies, jobs are intuitively linked to data analysis, which is also a meaning we're later going to want to assign them. Could we call this "track-dataset"?

Reply:
I am changing it to "track-upload". The full command would be "austrakka project dataset track-upload"